hooks/useDetectOS.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { detectOS } from '../util/detectOS';
+import { getBitness } from '../util/getBitness';
+import type { UserOS } from '../types/userOS';
+
+export const useDetectOS = () => {
+  const [userOSData, setUserOSData] = useState<{
+    os: UserOS;
```

Comment:
this should be a type (the type can be defined on top of the file)

Reply:
TBH, naming things in detect OS hook & util function is hard. 🤣 
FYI, I am going to change:
1. `type UserOS` to `type OS`
2. Name this new structure to be `UserOS`, "data"/"info" as suffix is unnecessary IMO...

Reply:
This should just be named UserOSState or something like that